rust/agama-server/src/network/nm/dbus.rs

```diff
@@ -1656,6 +1751,51 @@ mod test {
         assert_eq!(wep_key0, "5b73215e232f4c577c5073455d");
         let wep_key1: &str = security.get("wep-key1").unwrap().downcast_ref().unwrap();
         assert_eq!(wep_key1, "hello");
+
+        let group_algorithms: &zvariant::Array =
+            security.get("group").unwrap().downcast_ref().unwrap();
+        let group_algorithms: Vec<GroupAlgorithm> = group_algorithms
+            .get()
+            .iter()
+            .map(|x| x.downcast_ref::<str>().unwrap())
```

Comment:
Those `unwraps` are dangerous because is something goes wrong, the whole service would go down. Perhaps you might want to split this chain into smaller functions.

Reply:
This is the testing code

Reply:
Ouch, Friday afternoon, you know 😅